lib/uuid.c

```diff
@@ -89,8 +89,15 @@ M0_INTERNAL int m0_uuid_parse(const char *str, struct m0_uint128 *val)
 	    parse_hex(&str[14],  4, &h3) < 0 || str[18] != '-' ||
 	    parse_hex(&str[19],  4, &h4) < 0 || str[23] != '-' ||
 	    parse_hex(&str[24], 12, &h5) < 0 || str[36] != '\0')
-		return M0_ERR(-EINVAL);
-	val->u_hi = h1 << 32 | h2 << 16 | h3;
+	    
```

Comment:
We have the following statements and expressions here (simplified):

```c
if (expr1)
if (expr2)
  return err;
```
Usually, the logical AND operator is used for that:

```c
if (expr1 && expr2)
  return err;
```

Reply:
Hi Ivan,
Sorry for the confusion caused due to incorrect indentation I have the following scenario
`if (expr1){
     if (expr2){
        return err;
     }
}
`
So the logical && operator is not the intended operation I have updated the same in the recent commit